src/script_interface/system/System.cpp

```diff
@@ -111,6 +111,7 @@ System::System() : m_instance{}, m_leaves{std::make_shared<Leaves>()} {
            if (not(new_value > Utils::Vector3d::broadcast(0.))) {
              throw std::domain_error("Attribute 'box_l' must be > 0");
            }
+           m_instance->veto_boxl_change();
            m_instance->box_geo->set_length(new_value);
            m_instance->on_boxl_change();
```

Comment:
Note to reviewers: there are long-term plans to replace this clumsy `veto/on_change` mechanism by a proper message-passing interface (#4762).